mailpoet/lib/Subscription/Pages.php

```diff
@@ -230,14 +249,23 @@ public function confirm() {
   public function unsubscribe(string $method): void {
     if (
       !$this->isPreview()
-      && ($this->subscriber !== null)
-      && ($this->subscriber->status !== SubscriberEntity::STATUS_UNSUBSCRIBED)
+      && (!is_null($this->subscriber))
+      && ($this->subscriber->getStatus() !== SubscriberEntity::STATUS_UNSUBSCRIBED)
```

Comment:
Wow. So this was working thanks to the [BC code we added to the entity](https://github.com/mailpoet/mailpoet/blob/926620e8f82ca691f1c11f6e973a2f86a6117522/mailpoet/lib/Entities/SubscriberEntity.php#L193-L203). Nice catch! Hmm maybe we should consider starting throwing deprecation warnings inside the BC code.

Reply:
I agree with the deprecation warning idea! we may have this issue in other parts of the code base or may even introduce in future.